main/src/ca/uwaterloo/flix/language/phase/jvm/BackendObjType.scala

```diff
@@ -302,6 +304,57 @@ object BackendObjType {
 
   //case class Enum(sym: Symbol.EnumSym, args: List[BackendType]) extends BackendObjType
 
+  case object Tagged extends BackendObjType with Generatable {
+    def genByteCode()(implicit flix: Flix): Array[Byte] = {
+      val cm = ClassMaker.mkAbstractClass(this.jvmName)
+
+      cm.mkConstructor(Constructor)
+
+      cm.mkField(NameField)
+
+      cm.closeClassMaker()
+    }
+
+    def NameField: InstanceField = InstanceField(this.jvmName, IsPublic, NotFinal, NotVolatile, "tag", String.toTpe)
```

Comment:
Perhaps final?

Reply:
It's created like
```
let t = new Tag()
t.name = something
t.value = somethingElse
```
So it can't be final

Reply:
OK